packages/flutter_tools/lib/src/ios/simulators.dart

```diff
@@ -531,7 +532,8 @@ class IOSSimulator extends Device {
 
   Future<bool> _buildAndInstallApplicationBundle(ApplicationPackage app) async {
     // Step 1: Build the Xcode project.
-    bool buildResult = await buildIOSXcodeProject(app, buildForDevice: false);
+    // The build mode for the simulator is always debug.
```

Comment:
Perhaps we should return an error condition if the user tries `flutter run --release` on a simulator (instead of remapping to debug mode)?